pubsub/rabbitpubsub/rabbit.go

```diff
@@ -118,8 +121,25 @@ func (o *URLOpener) OpenTopicURL(ctx context.Context, u *url.URL) (*pubsub.Topic
 
 // OpenSubscriptionURL opens a pubsub.Subscription based on u.
 func (o *URLOpener) OpenSubscriptionURL(ctx context.Context, u *url.URL) (*pubsub.Subscription, error) {
-	for param := range u.Query() {
-		return nil, fmt.Errorf("open subscription %v: invalid query parameter %q", u, param)
```

Comment:
You should clone SubscriptionOptions here, and modify and use the clone below, otherwise setting a URL parameter on one OpenSubscriptionURL will affect the next OpenSubscriptionURL as well.

Reply:
Ok I follow what natspubsub is doing